### DISCUSSION TITLE
Authentication Flow Redirection Issue on iOS with signInWithOAuth Button

### DISCUSSION THREAD

**Discussion by brocchirodrigo** [supabase/supabase#15220] [Questions]
Good afternoon, everyone! I stumbled upon a problem and I need some help, but first of all, I would like to say that I found Supabase services fantastic!

Regarding the issue:
I'm using the latest version of NextJS and the Next Helpers package from Supabase.
I have an authentication page that uses the Auth package buttons (signInWithOtp and signInWithOAuth).
The email authentication button works perfectly on all platforms.
The problem I'm facing is with the signInWithOAuth button on iPhones and iPads. When the user tries to authenticate using Google or Twitter providers on Android or the web, the redirection and authentication flow work normally. However, on iOS, the redirection doesn't happen.
I've tried a few approaches, but without success, and I need help with this case.

Versions:
"@supabase/auth-helpers-nextjs": "^0.7.2",
"@supabase/supabase-js": "^2.26.0" (not currently in use on the pages),
"next": "13.4.6" (using the app folder).

**Comment by danielisfine1** [comment]
What was causing my issue was the use of router.refresh() in the error checking part of my handler. Interestingly, it was only causing the error in ios. 

  const loginWithGoogle = async () => {

    const { data, error } = await supabase.auth.signInWithOAuth({
      provider: 'google',
      options: {
        redirectTo: `${location.origin}/auth/callback`,
        queryParams: {
          access_type: 'offline',
          prompt: 'consent',
        },
      }
    });

    if (error !== null) {
      console.log(error?.message);
      return "error";
    } else {
      console.log(data);
      **_router.refresh();_**
      return "success";
    };    

  };
